src/index.ts

```diff
@@ -1,20 +1,28 @@
 import { h } from 'hastscript'
 import type { ShikiTransformer } from '@shikijs/types'
+import type { HTMLAttributes } from 'astro/types'
```

Comment:
The type `HTMLButtonElement` should be defined globally by TypeScript.
If not, we probably need to adjust the `tsconfig.json` to add `"dom"` to the `compilerOptions.lib` (see: https://stackoverflow.com/a/42604094)